WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanFragment.kt

```diff
@@ -81,6 +84,15 @@ class ScanFragment : Fragment(R.layout.scan_fragment) {
         }
     }
 
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == RequestCodes.SHOW_THREAT_DETAILS &&
+            data?.getBooleanExtra(ThreatDetailsFragment.REQUEST_SCAN_STATE, false) == true
+        ) {
+            viewModel.fetchScanState()
```

Comment:
🔍 Nitpick: We might want to consider adding "onScanStateRequested" method into the VM instead of calling an action (fetch) from the view layer. It's not a big deal in this case, but I think it's good practice so other devs won't feel tempted to start calling actions from the view layer even in other cases.

```
if (requestCode == RequestCodes.SHOW_THREAT_DETAILS &&
            data?.getBooleanExtra(ThreatDetailsFragment.REQUEST_SCAN_STATE, false) == true
        ) {
            viewModel.onScanStateRequested()
}
```

```
fun onScanStateRequested() {
     fetchScanState()
}
```

Reply:
Good point. In 2d0991d.